4337/contracts/Safe4337Module.sol

```diff
@@ -175,13 +182,17 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
             userOp.callGasLimit,
             userOp.maxFeePerGas,
             userOp.maxPriorityFeePerGas,
+            signatureTimestamps,
             entryPoint
         );
 
-        try ISafe(payable(userOp.sender)).checkSignatures(operationHash, "", userOp.signature) {
-            validationData = SIG_VALIDATION_SUCCESS;
+        // The timestamps are validated by the entry point, therefore we will not check them again
+        uint48 validAfter = uint48(signatureTimestamps >> 48);
+        uint48 validUntil = uint48(signatureTimestamps);
+        try ISafe(payable(userOp.sender)).checkSignatures(operationHash, "", signatures) {
+            validationData = _packValidationData(false, validUntil, validAfter);
         } catch {
-            validationData = SIG_VALIDATION_FAILED;
+            validationData = _packValidationData(true, validUntil, validAfter);
```

Comment:
I would not return the `validUntil, validAfter` in the signature error here for two reasons:
1. The signature is incorrect, so you don’t actually know what the real validity is
2. An invalid signature is invalid always, so the validity doesn’t really conceptually make sense IMO.

I think both are handled identically by the `EntryPoint` contract, so there is no real reason to pay the (minor) additional gas overhead of computing the shared signature.

Reply:
The reason why I decided to return it in both cases is:
1. This is how it was implemented in the reference contracts, and also, the reference function has a boolean for signature failure for some reason.
2. They may matter for simulation (erc4337 simulation is a black box, but for example, they don't want you to revert in `validateUserOp`). Looking at the simulation methods in the contract, the timestamps are being used: https://github.com/eth-infinitism/account-abstraction/blob/674b1f51164e641a18d5c141895bab9c96a68e9d/contracts/core/EntryPointSimulations.sol#L87-L123